pkg/scalers/honeycomb_scaler.go

```diff
@@ -0,0 +1,300 @@
+	}, nil
+}
+
+func parseHoneycombMetadata(config *scalersconfig.ScalerConfig) (honeycombMetadata, error) {
+	meta := honeycombMetadata{}
+	err := config.TypedConfig(&meta)
+	if err != nil {
+		return meta, fmt.Errorf("error parsing honeycomb metadata: %w", err)
+	}
+	meta.TriggerIndex = config.TriggerIndex
+
+	// Use queryRaw if provided, else build query from legacy fields
+	if raw, ok := config.TriggerMetadata["queryRaw"]; ok && raw != "" {
+		var q map[string]interface{}
+		if err := json.Unmarshal([]byte(raw), &q); err != nil {
+			return meta, fmt.Errorf("error parsing queryRaw: %w", err)
+		}
+		meta.Query = q
+	} else if meta.Query == nil {
+		q := make(map[string]interface{})
+		if len(meta.Breakdowns) > 0 {
+			q["breakdowns"] = meta.Breakdowns
+		}
+		if meta.Calculation != "" {
+			q["calculations"] = []map[string]string{{"op": meta.Calculation}}
+		}
+		if meta.Limit > 0 {
+			q["limit"] = meta.Limit
+		}
+		if meta.TimeRange > 0 {
+			q["time_range"] = meta.TimeRange
+		}
+		meta.Query = q
+	}
+	if meta.Query == nil {
+		return meta, errors.New("no valid query provided in 'queryRaw', 'query', or legacy fields")
+	}
+	return meta, nil
+}
+
+func (s *honeycombScaler) Close(context.Context) error { return nil }
+
+// ----- Core Query Logic -----
+
+func (s *honeycombScaler) executeHoneycombQuery(ctx context.Context) (float64, error) {
+	// 1. Create Query
+	createURL := fmt.Sprintf("%s/queries/%s", honeycombBaseURL, s.metadata.Dataset)
+	bodyBytes, err := json.Marshal(s.metadata.Query)
+	if err != nil {
+		return 0, fmt.Errorf("error marshaling Honeycomb create query body: %w", err)
+	}
+	req, err := http.NewRequestWithContext(ctx, "POST", createURL, bytes.NewBuffer(bodyBytes))
+	if err != nil {
+		return 0, fmt.Errorf("error creating Honeycomb create query request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Honeycomb-Team", s.metadata.APIKey)
+	resp, err := s.httpClient.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("honeycomb create query error: %w", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 && resp.StatusCode != 201 {
+		body, _ := io.ReadAll(resp.Body)
+		return 0, fmt.Errorf("honeycomb createQuery status: %s - %s", resp.Status, string(body))
+	}
+	var createRes struct {
+		ID string `json:"id"`
+	}
+	if err := json.NewDecoder(resp.Body).Decode(&createRes); err != nil {
+		return 0, fmt.Errorf("decode createQuery: %w", err)
+	}
+	if createRes.ID == "" {
+		return 0, errors.New("createQuery: missing query id")
+	}
+
+	// 2. Run Query
+	runURL := fmt.Sprintf("%s/query_results/%s", honeycombBaseURL, s.metadata.Dataset)
+	runBody, err := json.Marshal(map[string]interface{}{
+		"query_id":                   createRes.ID,
+		"disable_series":             false,
+		"disable_total_by_aggregate": true,
+		"disable_other_by_aggregate": true,
+		// Query results limit is 10000, see https://api-docs.honeycomb.io/api for details
+		"limit": honeycombQueryResultsLimit,
```

Comment:
Does it make sense to expose this to users? I mean, the limit must be enforced, but maybe a user prefers to query X items instead of the limit. Maybe we can expose this to users and check that value doesn't pass the limit.
Or maybe this is not a valid use case, saying the truth, I don't have any experience with honeycomb